app/models/user.rb

```diff
@@ -129,6 +130,23 @@ class User < ActiveRecord::Base
   }
   scope :admins, -> { where(admin: true) }
   scope :trainee, -> { where(trainee: true) }
+  scope :order_by_counts, -> (order_by, direction) {
+    unless order_by.in?(VALID_SORT_COLUMNS)
+      raise ArgumentError, "Attribute not allowed: #{order_by}"
+    end
+
+    if order_by == "report" && direction == "asc"
+      User.left_outer_joins(:reports).group("users.id").order(Arel.sql("count(reports.id) asc"))
+    elsif order_by == "report" && direction == "desc"
+      User.left_outer_joins(:reports).group("users.id").order(Arel.sql("count(reports.id) desc"))
+    elsif order_by == "comment" && direction == "asc"
+      User.left_outer_joins(:comments).group("users.id").order(Arel.sql("count(comments.id) asc"))
+    elsif order_by == "comment" && direction == "desc"
+      User.left_outer_joins(:comments).group("users.id").order(Arel.sql("count(comments.id) desc"))
+    else
+      User.order(order_by + " " + direction)
```

Comment:
directionはチェックしていないのでSQL Injectionできてしまいそうです。
directionに任意のSQLにして実行できてしまいます。

あと文字列結合は基本的には式展開使ったほうがよいです。

```suggestion
      User.order("#{order_by} #{direction}")
```

Reply:
directionの方を忘れてました...
式展開の方も、修正しました。